k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -1849,7 +1856,7 @@ private void moveDangerousMessageToSuspiciousFolderAndUpdateStatus(
             suspiciousFolder.open(Folder.OPEN_MODE_RO);
             LocalMessage movedMessage = suspiciousFolder.getMessage(newUid);
             updateStatus(account, folder, suspiciousFolder, progress, newMessages, todo,
-                    movedMessage, movedMessage, true, messagesToNotify, storageEditor);
+                    movedMessage, movedMessage, localMessage.getMessageId(), true, messagesToNotify, storageEditor);
```

Comment:
while this is different, you use the localMessage variable passed by the caller while you pass movedMessage as an object. Makes sense

Reply:
Yes, as I said above the issue is that the "movedMessage" variable could be null if there was some mismatch between remote and local folder. But thankfully that only happens in corner cases like removing an account etc.